lib/stdlib/src/zip.erl

```diff
@@ -822,6 +822,26 @@ add_cwd(_CWD, {_Name, _} = F) -> F;
 add_cwd("", F) -> F;
 add_cwd(CWD, F) -> filename:join(CWD, F).
 
+%% remove ".." components from the path to protect from directory traversal
+protect_from_traversal("") -> "";
+protect_from_traversal(F) ->
+	SafeName = filename:join(lists:filter(fun(E) -> E =/= ".." end, filename:split(F))),
+	%% filename:split/1 removes trailing separators so we append them here if needed
+	maybe_append_slash(F, SafeName, os:type()).
+
+maybe_append_slash(F, SafeName, {win32, _}) ->
+	IsDirectory = [lists:last(F)] == "\\",
```

Comment:
There is lists:suffix/2 for you. And case…end is often preferred instead of if…end for simple cases like these ones